product.py

```diff
@@ -23,6 +25,112 @@ class Product:
     "Product extension for Nereid"
     __name__ = "product.product"
 
+    display_available_quantity = fields.Boolean(
+        "Display Available Quantity On Website?"
+    )
+
+    start_displaying_qty_digits = fields.Function(
+        fields.Integer('Start Quantity Digits'),
+        getter='on_change_with_start_displaying_qty_digits'
+    )
+
+    start_displaying_available_quantity = fields.Numeric(
+        'Start Quantity', digits=(16, Eval('start_displaying_qty_digits', 2)),
+        states={
+            'invisible': ~Bool(Eval('display_available_quantity')),
+        }, depends=[
+            'display_available_quantity', 'start_displaying_qty_digits'
+        ],
+        help=(
+            "Product's available quantity must be less than this to show on"
+            " website"
+        )
+    )
+
+    min_warehouse_quantity = fields.Numeric(
+        'Min Warehouse Quantity', digits=(16, 4),
+        help="Minimum quantity required in warehouse for orders"
+    )
+
+    is_eligible_to_buy = fields.Function(
+        fields.Boolean("Is Eligible For Purchase?"),
+        getter="get_is_eligible_to_buy"
+    )
+
+    show_quantity_on_website = fields.Function(
+        fields.Boolean("Show Quantity on Website?"),
+        getter="get_show_quantity_on_website"
+    )
+
+    @classmethod
+    def __setup__(cls):
+        super(Product, cls).__setup__()
+
+        cls._error_messages.update({
+            'start_displaying_positive': (
+                'This quantity should be always be positive'
+            ),
+        })
+
+    @classmethod
+    def validate(cls, records):
+        """
+        Validation method
+        """
+        super(Product, cls).validate(records)
+
+        for record in records:
+            record.validate_start_display_quantity()
+
+    def validate_start_display_quantity(self):
+        """
+        This method validates that `start_displaying_available_quantity` is
+        always positive.
+        """
+        if self.start_displaying_available_quantity and \
+                self.start_displaying_available_quantity <= 0:
+            self.raise_user_error('start_displaying_positive')
+
+    @staticmethod
+    def default_min_warehouse_quantity():
+        """
+        By default, min_warehouse_quantity is zero. This is to handle the normal
+        sale order workflow.
+        """
+        return 0
+
+    @fields.depends('_parent_template')
+    def on_change_with_start_displaying_qty_digits(self, name=None):
+        """
+        Getter for start_displaying_qty_digits
+        """
+        return self.template.default_uom.digits or 2
+
+    def get_is_eligible_to_buy(self, name=None):
+        """
+        Getter for is_eligible_to_buy. This field is for inventory checking
+        purpose.
+        """
+        if self.display_available_quantity and self.warehouse_quantity:
```

Comment:
both are not connected. The quantity maynot be displayed and you might just want to say you cannot buy this because there is no inventory.